ui/components/component-library/button-icon/button-icon.types.ts

```diff
@@ -0,0 +1,49 @@
+import type { BoxProps } from '../../ui/box/box.d';
+import { IconName, IconSize } from '../icon';
+import { Color, IconColor } from '../../../helpers/constants/design-system';
+
+export enum ButtonIconSize {
+  Sm = IconSize.Sm,
+  Lg = IconSize.Lg,
+}
```

Comment:
Let's just use static strings until we are able to find a way to map or reuse enums
```suggestion
export enum ButtonIconSize {
  Sm = 'sm',
  Lg = 'lg',
}
```